sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/ExtractKeyPhraseAsyncClient.java

```diff
@@ -74,73 +75,78 @@
             Objects.requireNonNull(document, "'document' cannot be null.");
             final TextDocumentInput textDocumentInput = new TextDocumentInput("0", document);
             textDocumentInput.setLanguage(language);
-            return extractKeyPhrases(Collections.singletonList(textDocumentInput), null)
-                    .map(keyPhraseResult -> {
-                        if (keyPhraseResult.isError()) {
-                            throw logger.logExceptionAsError(toTextAnalyticsException(keyPhraseResult.getError()));
+            return extractKeyPhrasesWithResponse(Collections.singletonList(textDocumentInput), null)
+                    .map(keyPhraseResultCollection -> {
+                        for (ExtractKeyPhraseResult keyPhraseResult : keyPhraseResultCollection.getValue()) {
+                            if (keyPhraseResult.isError()) {
+                                throw logger.logExceptionAsError(toTextAnalyticsException(keyPhraseResult.getError()));
+                            }
+                            return new KeyPhrasesCollection(keyPhraseResult.getKeyPhrases(),
+                                keyPhraseResult.getKeyPhrases().getWarnings());
                         }
-                        return new KeyPhrasesCollection(keyPhraseResult.getKeyPhrases(),
-                            keyPhraseResult.getKeyPhrases().getWarnings());
-                    }).last();
+                        // this step should never be executed, if executed, we get an empty collection above.
+                        throw logger.logExceptionAsError(
+                            new TextAnalyticsException("None key Phrases extracted.",
```

Comment:
Is this something new we are introducing in this PR?
Rather if `keyPhraseResultCollection` is empty return the same to client-side?
What do other languages do here?

Reply:
Yes. I will update it to return empty.